lib/typhoeus/normalized_header_hash.rb

```diff
@@ -52,7 +52,7 @@ def delete(key)
 
   private
     def convert_key(key)
-      key.to_s.split(/_|-/).map { |segment| segment.capitalize }.join("-")
+      key.to_s.tr('_'.freeze,'-'.freeze).split('-'.freeze).map { |segment| segment.capitalize.freeze }.join('-'.freeze)
```

Comment:
I think there's even more room for improvement

We could extract `'_'.freeze` and `'-'.freeze` into constants and reuse them